jmxtrans-core/src/main/java/com/googlecode/jmxtrans/model/Result.java

```diff
@@ -52,18 +51,20 @@
 	@Getter private final String className;
 	@Getter private final String objDomain;
 	@Getter private final String typeName;
-	@Nonnull @Getter private final ImmutableMap<String, Object> values;
+	@Getter private final ImmutableList<String> valuePath;
+	@Nonnull @Getter private final Object value;
```

Comment:
While the values Map was ensured to be non null, I'm pretty sure that a single value can be null...

Reply:
The JmxResultProcessor throughs away null values so I left @Nonnull but that may not be good idea.

Reply:
If we do throw away null values, then `@NonNull` is fine. Though I now wonder if it is a good thing or not...